cfme/tests/physical_infrastructure/api/test_physical_server_api.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils import error
+from cfme.utils.rest import assert_response
+from cfme.utils.wait import wait_for
+from manageiq_client.api import APIException
+
+pytestmark = [
+    pytest.mark.tier(3),
+    pytest.mark.provider([LenovoProvider], scope='module')
+]
+
+# The max time that a single test has to perform the entire cycle of action
+# in seconds (5 min)
+TIMEOUT = 600
+
+# The time interval to perform a verification of state change
+# in seconds (1 min)
+DELAY = 60
+
+
+@pytest.fixture(scope="module")
+def physical_server(setup_provider_modscope, appliance):
+    physical_server = appliance.rest_api.collections.physical_servers[0]
+    return physical_server
+
+
+def get_server_attr(attr_name, provider, physical_server):
+    provider.refresh_provider_relationships()
+    physical_server.reload()
+    return physical_server[attr_name]
+
+
+def test_get_physical_server(physical_server, appliance):
+    existent_server = appliance.rest_api.get_entity('physical_servers', physical_server.id)
+    assert existent_server is not None
```

Comment:
REQUIRED: this will not fail even when the resource with id of ``physical_server.id`` doesn't exist. The ``existent_server`` will be an ``Entity`` instance, not ``None``. You need to do the same thing as you do in the ``test_get_nonexistent_physical_server``:
```python
existent_server.reload()  # it will fail here if the resource doesn't exist
assert_response(appliance)
```

Reply:
Hmm, good point! Thank you @mkoura